addon/mixins/data-adapter-mixin.js

```diff
@@ -56,6 +57,44 @@ export default Ember.Mixin.create({
   authorizer: null,
 
   /**
+    First authorizes the request and then calls base class' `ajax()`.
+
+    @param {String} url The requested URL.
+    @param {String} type The method of the HTTP request.
+    @param {Object} options Additional XHR options.
+  */
+  ajax(url, type, options) {
+    // tests benefit from doing it like this (instead of just `options = options || {}`).
+    if (!options) {
+      options = {};
+    }
+    const authorizer = this.get('authorizer');
+    Ember.assert("You're using the DataAdapterMixin without specifying an authorizer. Please add `authorizer: 'authorizer:application'` to your adapter.", Ember.isPresent(authorizer));
+
+    // Because of the way _super() works, we need to store references to base
+    // class implementation here in order to be able to use it in the callback below.
+    const me = this;
+    const supr = this._super;  // for ember 2.x
+    const nextsuper = this.__nextSuper;  // for ember 1.x
```

Comment:
No, didn't have the time yet. There still seem to be some problems with 1.1 which need to be fixed before I can take on 1.2 things.